src/models/transactions/common.ts

```diff
@@ -0,0 +1,136 @@
+import { ValidationError } from "../../common/errors"
+import { Memo, Signer } from "../common"
+
+const transactionTypes = [
+    "AccountSet",
+    "AccountDelete",
+    "CheckCancel",
+    "CheckCash",
+    "CheckCreate",
+    "DepositPreauth",
+    "EscrowCancel",
+    "EscrowCreate",
+    "EscrowFinish",
+    "OfferCancel",
+    "OfferCreate",
+    "Payment",
+    "PaymentChannelClaim",
+    "PaymentChannelCreate",
+    "PaymentChannelFund",
+    "SetRegularKey",
+    "SignerListSet",
+    "TicketCreate",
+    "TrustSet",
+]
+
+const isMemo = (obj: {Memo: Memo}): boolean => {
+    const memo = obj.Memo
+    const size = Object.keys(memo).length
+    const validData = memo.MemoData === undefined 
+        || typeof memo.MemoData === 'string'
+    const validFormat = memo.MemoFormat === undefined 
+        || typeof memo.MemoData === 'string'
+    const validType = memo.MemoType === undefined
+        || typeof memo.MemoType === 'string'
+
+    const otherKeys = Object.keys(memo).every((key: string) => {
+        return key === "MemoFormat" 
+            || key === "MemoData"
+            || key === "MemoType"
+    })
+
+    return (1 <= size && size <= 3) 
+        && validData 
+        && validFormat 
+        && validType
+        && otherKeys
+}
+
+const isSigner = (signer: Signer): boolean => {
+    return Object.keys(signer).length === 3
+        && typeof signer.Account === 'string'
+        && typeof signer.TxnSignature === 'string'
+        && typeof signer.SigningPubKey === 'string'
+}
+
+export interface CommonFields {
+  Account: string;
+  TransactionType: string;
+  Fee?: string;
+  Sequence?: number;
+  AccountTxnID?: string;
+  Flags?: number;
+  LastLedgerSequence?: number;
+  Memos?: Array<{ Memo: Memo }>;
+  Signers?: Array<Signer>;
+  SourceTag?: number;
+  SigningPubKey?: string;
+  TicketSequence?: number;
+  TxnSignature?: string;
+}
+
+/**
+ * verify the common fields of a transaction. The verify functionality will be
+ * optional, and will check transaction form at runtime. This should be called
+ * any time a transaction will be verified.
+ * 
+ * @param common - An interface w/ common transaction fields
+ * @returns - Void
+ * @throws - When the common param is malformed. 
+ */
+export function verifyCommonFields(common: CommonFields): void {
+    if (common.Account === undefined)
```

Comment:
If y'all are ok with it, I would prefer to avoid these kinds of runtime checks and just lean on rippled to catch such errors. In practice, developers don't need us to provide this kind of verify code. rippled is the source of truth, and ideally, it's the single source of truth.

Reply:
TypeScript will also catch some of these mistakes (if the developer is using TS; JavaScript developers can still use this library -- without type safety -- too)

Reply:
Yeah, I'm in favor of not checking anything that TS will check automatically

Reply:
That's fair, what should we verify in `verify`? Should it be preflight checks, like elliot suggested? 

IMO, we can check almost everything w/ typescript, I was leaving this functionality in for JS devs.

Reply:
@natenichols and I chatted about this, we're going to leave the functionality in but make it optional (so the function will exist for those who want it, but you can also choose to not use it).